docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/browser-provider-methods.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: docs
+title: Browser Provider Methods
+permalink: /documentation/extending-testcafe/custom-browser-provider-plugin/browser-provider-methods.html
+checked: false
+---
+# Browser Provider Methods
+
+To create your [custom browser provider](index.md#implementing-the-browser-provider), implement the following provider methods.
+
+## Required Methods
+
+### openBrowser
+
+Runs the specified browser and opens the specified page in it.
+
+```text
+openBrowser(id, browserName, startPage)
+```
+
+Parameter     | Type   | Description
+------------- | ------ | ----------------------------
+`id`          | String | The browser identifier.
+`browserName` | String | The browser name.
+`startPage`   | String | The url of the page to open.
+
+See [Example](index.md#example).
+
+### closeBrowser
+
+Closes the specified browser.
+
+```text
+closeBrowser(id, pageInfo)
+```
+
+Parameter   | Type   | Description
+----------- | ------ | -----------------------------------
+`id`        | String | The browser identifier.
+`pageInfo`  | Object | The [pageInfo](#pageinfo) object.
+
+#### pageInfo
+
+The `pageInfo` object provides access to a certain web page and has the following properties.
+
+Property  | Type   | Description
+--------- | ------ | ------------------
+`title`   | String | The page title.
+`url`     | String | The page URL.
+`width`   | Number | The page swidth, in pixels.
+`height`  | Number | The page height, in pixels.
+
+See [Example](index.md#example).
+
+## Optional Methods
+
+### init
+
+Performs initialization actions the provider may require.
```

Comment:
It is called only once in `testcafe` process lifetime before any call to other provider API methods will be made. It may be used to initialize required data in the provider.

`dispose` is similarly called once before `testcafe` process will be finished, you can use it to perform clean-up of your provider data.